Migrate backends/apple/coreml to the new namespace

dbort opened the pull request:
Summary: Move the Core ML backend out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Differential Revision: D63995558

cccclai commented:
@cymbalrush can you take a look?

cccclai reviewed (approved):
Looks good if CI passes

cymbalrush reviewed (approved):
Thank you!

YifanShenSZ reviewed (approved):
Thank you 🙏

dbort committed:
Migrate backends/apple/coreml to the new namespace (#5943)

Summary:

Move the Core ML backend out of the `torch::` namespace, and update to avoid using the `torch::` or `exec_aten::` namespaces.

Reviewed By: cccclai

Differential Revision: D63995558

dbort commented:
@pytorchbot cherry-pick --onto release/0.4 -c fixnewfeature
